cli/internal/backends/backends.go

```diff
@@ -9,21 +9,24 @@ import (
 
 var backends = []api.LanguageBackend{
 	nodejs.NodejsYarnBackend,
+	nodejs.NodejsBerryBackend,
 	nodejs.NodejsNpmBackend,
 	nodejs.NodejsPnpmBackend,
 }
 
 func GetBackend() (*api.LanguageBackend, error) {
 	for _, b := range backends {
 		if fs.FileExists(b.Specfile) &&
-			fs.FileExists(b.Lockfile) {
+			fs.FileExists(b.Lockfile) &&
+			b.FinalCheck() {
```

Comment:
can we refactor to be a single function called Detect() and then   Get rid of FinalCheck?

Reply:
So `b.FinalCheck()` to `b.Detect()` or the entire if check to `Detect()`?

Reply:
- `GetBackend()` should take are path as well ( and should become `GetBackend(cwd string)`)
- then the entire check should go into `Detect(cwd string)`  where `cwd` argument

Very soon, we want to tackle being able to run `turbo`  from a subdirectory and having it "just work" (https://github.com/vercel/turborepo/issues/144). So this `cwd` stuff will pay dividends

Reply:
Correct!